database/postgres/postgres.go

```diff
@@ -41,17 +41,40 @@ type Config struct {
 	StatementTimeout time.Duration
 }
 
+type SQLWrapper struct {
+	*sql.DB
+}
+
+func (w *SQLWrapper) Conn(ctx context.Context) (Conn, error) {
+	return w.DB.Conn(ctx)
+}
+
+type Conn interface {
+	Close() error
+	ExecContext(ctx context.Context, query string, args ...interface{}) (sql.Result, error)
+	BeginTx(ctx context.Context, opts *sql.TxOptions) (*sql.Tx, error)
+	QueryContext(ctx context.Context, query string, args ...interface{}) (*sql.Rows, error)
+	QueryRowContext(ctx context.Context, query string, args ...interface{}) *sql.Row
+}
+
+type DB interface {
+	Close() error
+	Ping() error
+	QueryRow(query string, args ...interface{}) *sql.Row
+	Conn(ctx context.Context) (Conn, error)
+}
+
 type Postgres struct {
 	// Locking and unlocking need to use the same connection
-	conn     *sql.Conn
-	db       *sql.DB
+	conn     Conn
+	db       DB
 	isLocked bool
 
 	// Open and WithInstance need to guarantee that config is never nil
 	config *Config
 }
 
-func WithInstance(instance *sql.DB, config *Config) (database.Driver, error) {
+func WithInterface(instance DB, config *Config) (database.Driver, error) {
```

Comment:
We should be able to keep the same method `WithInstance()` and just change the param to use the interface.